plugins-builtin/extractcode-libarchive-system_provided/src/extractcode_libarchive/__init__.py

```diff
@@ -28,36 +28,61 @@
 from os.path import abspath
 from os.path import dirname
 from os.path import join
+from os import path
 
 from plugincode.location_provider import LocationProviderPlugin
 from plugincode.location_provider import location_provider_impl
 
 import platform
 
 class LibarchivePaths(LocationProviderPlugin):
-	def get_locations(self):
-		curr_dir = dirname(abspath(__file__))
-		distribution=platform.linux_distribution()[0]	
-	
-		# List of various major distributions consisting of flavors
-		debian_based_distro=['Ubuntu','Mint','debian']
+    def get_locations(self):
+
+        '''This defines the various mainstream system available 
```

Comment:
We use triple double quotes """ for docstrings and no empty line before. We also use the imperative style and describe what is returned
So here this should be:
```
    def get_locations(self):
        """
        Return a mapping of {location key: location} providing the installation
        locations of the libarchive shared library as installed on various Linux
        distros or on FreeBSD.
        """
```

Reply:
Will take care of these things from next time.
And will try to improve my imperative style comment.

Reply:
Taken Note off!!
Will take care of these things from next time.
And will try to improve my imperative style comment.